source/StarField.h

```diff
@@ -17,6 +17,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #define STAR_FIELD_H_
 
 #include "Shader.h"
+#include "System.h"
```

Comment:
Use a forward declaration (`class System;`) here and #include system in the cpp file.